lib/SIL/SILFunctionType.cpp

```diff
@@ -245,29 +251,40 @@ CanSILFunctionType SILFunctionType::getAutoDiffAssociatedFunctionType(
                                DependentMemberType *dependentType) -> CanType {
     // Builtins are their own Tangent/Cotangent.
     if (type->is<BuiltinType>()) return type->getCanonicalType();
-    return dependentType->substBaseType(type, lookupConformance)
-        ->getCanonicalType();
+
+    auto assocTy = dependentType->substBaseType(type, lookupConformance);
```

Comment:
What are you trying to accomplish here?

Reply:
By the way, this particular change happened in https://github.com/apple/swift/pull/20980 so I'll be making the change there, and the change should vanish from this PR when I merge that one and rebase this one.